core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/pythonUDF/PythonUDFOpDesc.java

```diff
@@ -69,21 +91,32 @@ public OperatorInfo operatorInfo() {
     @Override
     public Schema getOutputSchema(Schema[] schemas) {
         Schema inputSchema = schemas[0];
-        if (inputColumns != null && !inputColumns.isEmpty()) {
+
+        // check if inputColumns are presented in inputSchema.
+        if (inputColumns != null) {
             for (String s : inputColumns) {
-                if (!inputSchema.containsAttribute(s)) throw new RuntimeException("No such column:" + s);
+                if (!inputSchema.containsAttribute(s)) throw new RuntimeException("No such column:" + s + ".");
             }
         }
 
-        Schema outputSchema = inputSchema;
-        if (outputColumns != null && !outputColumns.isEmpty()) {
+        Schema.Builder outputSchemaBuilder = Schema.newBuilder();
+        if (pythonUDFType != Training) {
+            outputSchemaBuilder.add(inputSchema);
+        } else {
+            outputSchemaBuilder.add("class", AttributeType.STRING);
```

Comment:
Looks good. I think this can be a good temporary solution until we add OpDesc support to python udf in the future. 
Since this is definitely short term hacky design rather than a long term solution, you can decide what attributes you put into here based on your need and the user's requirement.